tools/generate-mocks/generate-mocks.ts

```diff
@@ -113,44 +114,44 @@ ${providers},
     this.saveFileLinted(filename, content);
   }
 
-  private async getExportedTypes(
-    folderpath: string,
-    assertedTypesMap: Map<string, string>
-  ): Promise<string[]> {
+  private async getExportedTypesAndAliases(folderpath: string): Promise<string[]> {
     const files = await this.getBarrelFiles(folderpath);
-    const exportedTypes = await Promise.all(
+    const exportedTypesAndAliases = await Promise.all(
       files.map(async (file) => {
         const fileContent = await readFile(file, 'utf8');
-        const typesInFile = [];
-        const exportRegEx = /^export \{ *(.*) *\} from '\.\//;
-        const exportRegExGlobal = new RegExp(exportRegEx, 'gm');
-        // "|| []" prevents having to check for undefined:
-        (fileContent.match(exportRegExGlobal) || []).forEach((matchedLine) => {
-          // "slice(1)" skips the full match:
-          (matchedLine.match(exportRegEx) || []).slice(1).forEach((exported) => {
-            // Split multiple entries, trim away whitespace and add:
-            typesInFile.push(...exported.split(',').map((entry) => entry.trim()));
-          });
-        });
-        const filteredTypes = this.mapAndRemoveTypeAssertions(typesInFile, assertedTypesMap);
-        return filteredTypes;
+        return this.getTypesInFile(fileContent);
       })
     );
-    return Array.prototype.concat(...exportedTypes);
+    return Array.prototype.concat(...exportedTypesAndAliases);
   }
 
-  mapAndRemoveTypeAssertions(
-    exportedTypes: string[],
-    assertedTypesMap: Map<string, string>
+  private getTypesInFile(fileContent: any): any[] {
+    const typesInFile = [];
+    const exportRegEx = /^export \{ *(.*) *\} from '\.\//;
+    const exportRegExGlobal = new RegExp(exportRegEx, 'gm');
+    // "|| []" prevents having to check for undefined:
+    (fileContent.match(exportRegExGlobal) || []).forEach((matchedLine) => {
+      // "slice(1)" skips the full match:
+      (matchedLine.match(exportRegEx) || []).slice(1).forEach((exported) => {
+        // Split multiple entries, trim away whitespace and add:
+        typesInFile.push(...exported.split(',').map((entry) => entry.trim()));
+      });
+    });
+    return typesInFile;
+  }
+
+  mapAndRemoveAliases(
+    exportedTypesAndAliases: string[],
+    aliasesMap: Map<string, string>
```

Comment:
I'm not going to block if it is not changed - but I would still vouch for removing `aliasesMap` as an argument as it is not really needed to compute what has to be computed. aliasesMap is a result of `mapAndRemoveAliases`. 

Instead I would consider moving line 35: 
```Typescript
const aliasesMap = new Map<string, string>();
```

to line ~147: 
```Typescript
mapAndRemoveAliases(
    exportedTypesAndAliases: string[]
  ): string[] {
    const aliasesMap = new Map<string, string>();
    // capture group 1 is internal component name, group 2 is exported name
    const aliasRegex = /(\w+) as (\w+)/;
    ...
```

And then returning a tuple from `mapAndRemoveAliases`: 
```typescript
return [exports, aliasesMap]
```